acceptance: TestDockerCLI_test_demo_multitenant failed [close of closed channel in start_drpc_listener.go]

acceptance.TestDockerCLI_test_demo_multitenant [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_Acceptance/20476298?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_Acceptance/20476298?buildTab=artifacts#/) on master @ [277f3f584acad5332603a6e6b82bcd6e318249d5](https://github.com/cockroachdb/cockroach/commits/277f3f584acad5332603a6e6b82bcd6e318249d5):


```
=== RUN   TestDockerCLI_test_demo_multitenant
    test_log_scope.go:171: test logs captured to: /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/acceptance/_tmp/f14cb1cd56830b137df13910221b8384/logTestDockerCLI_test_demo_multitenant1205244811
    test_log_scope.go:82: use -show-logs to present logs inline
    generated_cli_test.go:126: -- test log scope end --
test logs left over in: /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/acceptance/_tmp/f14cb1cd56830b137df13910221b8384/logTestDockerCLI_test_demo_multitenant1205244811
--- FAIL: TestDockerCLI_test_demo_multitenant (29.93s)
=== RUN   TestDockerCLI_test_demo_multitenant/test_demo_multitenant.tcl
    util_docker.go:256: non-zero exit code: 1
--- FAIL: TestDockerCLI_test_demo_multitenant/test_demo_multitenant.tcl (18.94s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-foundations
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestDockerCLI_test_demo_multitenant.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-54513

--- a/pkg/server/start_drpc_listener.go
+++ b/pkg/server/start_drpc_listener.go
@@ -44,23 +44,3 @@ func (ln *dropDRPCHeaderListener) Close() error {
 func (ln *dropDRPCHeaderListener) Addr() net.Addr {
 	return ln.wrapped.Addr()
 }
-
-type noopListener struct{ done chan struct{} }
-
-func (l *noopListener) Accept() (net.Conn, error) {
-	<-l.done
-	return nil, net.ErrClosed
-}
-
-func (l *noopListener) Close() error {
-	if l.done == nil {
-		return nil
-	}
-	close(l.done)
-	l.done = nil
-	return nil
-}
-
-func (l *noopListener) Addr() net.Addr {
-	return nil
-}

--- a/pkg/server/start_listen.go
+++ b/pkg/server/start_listen.go
@@ -14,7 +14,6 @@ import (
 
 	"github.com/cockroachdb/cmux"
 	"github.com/cockroachdb/cockroach/pkg/rpc"
-	"github.com/cockroachdb/cockroach/pkg/rpc/rpcbase"
 	"github.com/cockroachdb/cockroach/pkg/sql/pgwire"
 	"github.com/cockroachdb/cockroach/pkg/util/log"
 	"github.com/cockroachdb/cockroach/pkg/util/netutil"
@@ -135,22 +134,10 @@ func startListenRPCAndSQL(
 		}
 	}
 
-	// Host drpc only if it's _possible_ to turn it on (this requires a test build
-	// or env var). If the setting _is_ on, then it was overridden in testing and
-	// we want to host the server too.
-	hostDRPC := rpcbase.ExperimentalDRPCEnabled.Validate(nil /* not used */, true) == nil ||
-		rpcbase.ExperimentalDRPCEnabled.Get(&cfg.Settings.SV)
-
-	// If we're not hosting drpc, make a listener that never accepts anything.
-	// We will start the dRPC server all the same; it barely consumes any
-	// resources.
-	var drpcL net.Listener = &noopListener{make(chan struct{})}
-	if hostDRPC {
-		// Throw away the header before passing the conn to the drpc server. This
-		// would not be required explicitly if we used `drpcmigrate.ListenMux` but
-		// cmux keeps the prefix.
-		drpcL = &dropDRPCHeaderListener{wrapped: m.Match(drpcMatcher)}
-	}
+	// dropDRPCHeaderListener discards the header before passing the connection
+	// to the drpc server. This would not be required if we used
+	// `drpcmigrate.ListenMux`, but cmux keeps the prefix.
+	var drpcL net.Listener = &dropDRPCHeaderListener{wrapped: m.Match(drpcMatcher)}
 
 	grpcL := m.Match(cmux.Any())
 	if serverTestKnobs, ok := cfg.TestingKnobs.Server.(*TestingKnobs); ok {

